.github/workflows/CI_build.yml

```
@@ -29,13 +29,26 @@ jobs:
       - name: Build package and create dev environment
         run: |
           python3 -m pip install --upgrade pip setuptools wheel
-          python3 -m pip install -e .[dev]
+          python3 -m pip install -e ".[dev]"
       - name: Check style against standards using prospector
         shell: bash -l {0}
         run: prospector --profile linter_profile -o grouped -o pylint:pylint-report.txt --zero-exit
-      - name: Run unit test / Test coverage with Scrutinizer
+      - name: Run all unit test with Tensorflow backend
+        run: |
+          python3 -m pip install -e ".[tensorflow]"
+          coverage run --source=mcfly --data-file=.coverage.tensorflow -m pytest
+      - name: Run JAX backend tests
+        run: |
+          python3 -m pip install -e ".[jax]"
+          keras_backend=jax coverage run --source=mcfly --data-file=.coverage.jax -m pytest -m jax
+      - name: Run PyTorch backend tests
+        run: |
+          python3 -m pip install --extra-index-url https://download.pytorch.org/whl/cpu -e ".[torch]"
+          keras_backend=torch coverage run --source=mcfly --data-file=.coverage.torch -m pytest -m torch
+      - name: Combine coverage reports
+        run: coverage combine --keep .coverage.tensorflow .coverage.jax .coverage.torch
+      - name: Upload test coverage with Scrutinizer
```

The results cannot be uploaded from a PR by an external contributor because there won't be access to the secret tokens. So only run this conditionally.
```suggestion
      - name: Upload test coverage with Scrutinizer
      if: github.event.pull_request.head.repo.full_name == github.repository
```